src/mono/mono/mini/method-to-ir.c

```diff
@@ -2317,7 +2317,10 @@ static void
 emit_invalid_program_with_msg (MonoCompile *cfg, MonoError *error_msg, MonoMethod *caller, MonoMethod *callee)
 {
 	g_assert (!is_ok (error_msg));
-	char *str = mono_mempool_strdup (cfg->domain->mp, mono_error_get_message (error_msg));
+	MonoMemoryManager *memory_manager = mono_domain_ambient_memory_manager (cfg->domain);
```

Comment:
Not sure why this didn't lock before.

Reply:
I assumed it was taking the domain lock at some higher level or was just an oversight.